NewsApi/routes.py

```diff
@@ -1,134 +1,102 @@
-def get_one_headline(id):
-    """
-    This function responds to a request for /get-one-news-headline/{news_headline_id}
-    with one matching news headline from the headlines.
-    :param id:   ID of news headline to find
-    :return:news headline matching ID
-    """
-    # Get the news_headline requested
-    news_headline = models.Headlines.query.filter(models.Headlines.id==id).one_or_none()
-
-    # Check if the requested id exists
-    if news_headline is not None:
-        # Serialize the data for the response
-        news_schema = models.HeadlinesSchema()
-        return make_response(jsonify(news_schema.dump(news_headline)), 201)
-    else:
-        return make_response(f'News Headline not found for the given Id: {id}', 404)
-
-
-
-@app.route('/get-everything/<query>')
-def get_everything(query):
-    """
-        This function takes data from newsapi and stores it in a database
-        """
-    newsapi = NewsApiClient(api_key=api_key)
-    everything = newsapi.get_everything(qintitle=query)
-    articles = everything['articles']
-    for i in range(len(articles)):
-        my_articles = articles[i]
-        found_articles = models.SearchedNews(author=my_articles['author'], content=my_articles['content'],
-                                         description=my_articles['description'], publishedAt=my_articles['publishedAt'],
-                                         source=my_articles['source'], title=my_articles['title'],
-                                         url=my_articles['url'], urlToImage=my_articles['urlToImage'])
-        db.session.add(found_articles)
-    db.session.commit()
-    return make_response(jsonify(response='OK'), 201)
-
-
-@app.route('/news-sentiment', methods=['GET'])
-def news_sentiment():
-    news = db.session.query(models.SearchedNews).all()
-    contents = [item.content for item in news]
-    text = ' '.join(contents)
-    senti = SentimentAnalyzer.sentiment(text)
-    score = SentimentAnalyzer.score(text)
-    #return make_response(jsonify(text=text), 201)
-    return make_response(jsonify(sentiment=senti, score=score), 201)
+    try:
+        body = request.get_json()
+        user = User(**body)
+
+        if len(body.get("password")) >= 8:
+            user.hash_password()
+        else:
+            return make_response(jsonify(message='Password should be at least 8 characters'), 400)
+
+        db.session.add(user)
+        db.session.commit()
+
+        return make_response(jsonify(message='New user added successfully'), 201)
+    except exc.SQLAlchemyError as e:
+        return make_response(jsonify(SQLAlchemy_error_message=str(e)), 400)
+    except Exception as e:
+        return make_response(jsonify(error_message=str(e)), 400)
+
 
+@app.route('/login', methods=['GET', 'POST'])
+def login():
+    try:
+        user = User.query.filter_by(email=request.json['email']).first()
 
+        if user is not None:
+            authorized = user.check_password(password=request.json['password'])
```

Comment:
I see that in this file you use both `body.get_json()` and `body["property"]`. You could go with one approach to keep consistency.